pkg/compare/slice.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package compare
+
+import (
+	"github.com/aws-controllers-k8s/runtime/pkg/util"
+)
+
+// SliceStringPEqual returns true if the supplied slices of string pointers
+// have equal values regardless of order.
+func SliceStringPEqual(a, b []*string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for _, aVal := range a {
+		if !util.InStringPs(*aVal, b) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
nit: i'm not sure if this is a big deal but this function can return true for `["a", "a", "b"]` and `["a", "b", "b"]` . Maybe worth to concider in the future

Reply:
ah, good point. I will update and add a test case for that.